File: shop/src/main/java/com/example/shop/entity/Address.java
package com.example.shop.entity;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "addresses")
public class Address {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "user_id", nullable = false)
	private User user;

	@Column(name = "receiver_name", nullable = false, length = 50)
	private String receiverName;

	@Column(nullable = false, length = 20)
	private String phone;

	@Column(nullable = false, length = 50)
	private String province;

	@Column(nullable = false, length = 50)
	private String city;

	@Column(nullable = false, length = 50)
	private String district;

	@Column(nullable = false, length = 200)
	private String detail;

	@Column(name = "is_default", nullable = false)
	private Boolean isDefault = false;

	@CreationTimestamp
	@Column(name = "created_at")
	private LocalDateTime createdAt;
}

File: shop/frontend/src/api/address.ts
import { api } from './index'

export interface Address {
  id: number
  receiverName: string
  phone: string
  province: string
  city: string
  district: string
  detail: string
  isDefault: boolean
  createdAt: string
  updatedAt: string
}

export interface AddressCreateRequest {
  receiverName: string
  phone: string
  province: string
  city: string
  district: string
  detail: string
  isDefault?: boolean
}

// 获取地址列表
export const getAddresses = () => {
  return api.get<Address[]>('/address/list')
}

// 创建地址
export const createAddress = (data: AddressCreateRequest) => {
  return api.post<Address>('/address/create', data)
}

// 删除地址
export const deleteAddress = (id: number) => {
  return api.post('/address/delete', { id })
}

// 更新地址
export const updateAddress = (data: Address & { id: number }) => {
  return api.put<Address>('/address/update', data)
}

// 设为默认
export const setDefaultAddress = (id: number) => {
  return api.post('/address/set-default', id)
}
